server/gitlab.go

```diff
@@ -0,0 +1,277 @@
+package server
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+const (
+	envKeyPRNumber = "PR_NUMBER"
+	envKeyBuildTag = "BUILD_TAG"
+)
+
+type GitLabClient struct {
+	client *gitlab.Client
+
+	Pipelines PipelinesService
+}
+
+func NewGitLabClient(accessToken string, baseURL string) (*GitLabClient, error) {
+	c, err := gitlab.NewClient(accessToken, gitlab.WithBaseURL(baseURL))
+	if err != nil {
+		return nil, err
+	}
+
+	return &GitLabClient{
+		client:    c,
+		Pipelines: c.Pipelines,
+	}, nil
+}
+
+// PipelinesService exposes an interface of GitLabCI client.
+// Useful to mock in tests.
+type PipelinesService interface {
+	CancelPipelineBuild(pid interface{}, pipeline int, options ...gitlab.RequestOptionFunc) (*gitlab.Pipeline, *gitlab.Response, error)
+	CreatePipeline(pid interface{}, opt *gitlab.CreatePipelineOptions, options ...gitlab.RequestOptionFunc) (*gitlab.Pipeline, *gitlab.Response, error)
+	GetPipeline(pid interface{}, pipeline int, options ...gitlab.RequestOptionFunc) (*gitlab.Pipeline, *gitlab.Response, error)
+	GetPipelineVariables(pid interface{}, pipeline int, options ...gitlab.RequestOptionFunc) ([]*gitlab.PipelineVariable, *gitlab.Response, error)
+	ListProjectPipelines(pid interface{}, opt *gitlab.ListProjectPipelinesOptions, options ...gitlab.RequestOptionFunc) ([]*gitlab.PipelineInfo, *gitlab.Response, error)
+}
+
+func (s *Server) triggerE2EGitLabPipeline(ctx context.Context, info *E2ETestTriggerInfo) (string, error) {
+	createOpts := &gitlab.CreatePipelineOptions{
+		Ref: &info.RefToTrigger,
+		Variables: []*gitlab.PipelineVariable{
+			{
+				Key:   "REF_MATTERMOST_WEBAPP",
+				Value: info.WebappBranch,
+			},
+			{
+				Key:   "SHA_MATTERMOST_WEBAPP",
+				Value: info.WebappSHA,
+			},
+			{
+				Key:   "REF_MATTERMOST_SERVER",
+				Value: info.ServerBranch,
+			},
+			{
+				Key:   "SHA_MATTERMOST_SERVER",
+				Value: info.WebappSHA,
+			},
+			{
+				Key:   envKeyBuildTag,
+				Value: info.BuildTag,
+			},
+			{
+				Key:   envKeyPRNumber,
+				Value: strconv.Itoa(info.TriggerPR),
+			},
```

Comment:
I can't see where it includes optional pipeline variables such as for `MM_ENV`, `INCLUDE_FILE` and `EXCLUDE_FILE`. It's assigned to [info.EnvVars](https://github.com/mattermost/mattermost-mattermod/pull/282/files#diff-7047d41fe350835623854ec21ede87cc3c32f0f8842eba189be55bfb1d459bc6R180) but didn't see any loop/append to `Variables: []*gitlab.PipelineVariable`.